### PR TITLE
Simplify bound params (Part 2/2)

### PR DESCRIPTION
**Pull request by msmk0**
Make the `.position(...)` accessor of the bound parameters take a geometry context as argument instead of using the default one. This is required to consistently compute the local-to-global transformation regardless of the surface type and detector conditions.

Must be merged after #327. Supersedes and closes #298.

### CODE REVIEW

**Comment by msmk0** [comment]
Ready to be reviewed and merged.

File: Core/include/Acts/EventData/TrackParametersConcept.hpp [hunk 1]
```diff
@@ -76,15 +80,16 @@ struct BoundTrackParametersConceptImpl {
                 "Missing/ invalid 'referenceSurface' method");
   static_assert(hasMethodParameters, "Missing/ invvalid 'parameters' method");
   static_assert(hasMethodCovariance, "Missing/ invvalid 'covariance' method");
```
Comment (asalzburger) [inline Core/include/Acts/EventData/TrackParametersConcept.hpp:82]: Should that really be `invvalid` ? Looks like you copied that naming, should change to `invalid` I suppose.

**Review by asalzburger — CHANGES_REQUESTED** [review]

**Comment by asalzburger** [comment]
Straight-forward follow-up of Part 1, if you can just change the `invvalid` then this can go in.

File: Core/include/Acts/EventData/SingleBoundTrackParameters.hpp [hunk 2]
```diff
@@ -8,16 +8,16 @@
...
-
 #include "Acts/EventData/ChargePolicy.hpp"
 #include "Acts/EventData/ParameterSet.hpp"
 #include "Acts/EventData/detail/PrintParameters.hpp"
 #include "Acts/EventData/detail/TransformationFreeToBound.hpp"
 #include "Acts/Surfaces/Surface.hpp"
 #include "Acts/Utilities/UnitVectors.hpp"
 
+#include <cassert>
+#include <cmath>
```
[the remark this answers is not in the document]
Comment (msmk0) [reply Core/include/Acts/EventData/SingleBoundTrackParameters.hpp:19]: Yes, the include ordering is handled automatically by clang-format.

File: Core/include/Acts/EventData/TrackParametersConcept.hpp (hunk 1 above)
Comment (msmk0) [reply Core/include/Acts/EventData/TrackParametersConcept.hpp]: Fixed.

**Review by asalzburger — APPROVED** [review]

**Review by asalzburger — APPROVED** [review]

**Pull request merged**